sdk/eventgrid/azure-resourcemanager-eventgrid/src/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java

```diff
@@ -136,15 +137,14 @@ private static void runSample() {
 
             // 4. Create an event grid subscription.
             EventSubscription eventSubscription = eventGridManager.topicEventSubscriptions()
-                .define(EVENT_SUBSCRIPTION_NAME)
-                .withExistingTopic(resourceGroup.name(), eventGridTopic.name())
-                .withDestination(new EventHubEventSubscriptionDestination()
-                    .withResourceId(eventHub.id()))
-                .withFilter(new EventSubscriptionFilter()
-                    .withIsSubjectCaseSensitive(false)
-                    .withSubjectBeginsWith("")
-                    .withSubjectEndsWith(""))
-                .create();
+                .createOrUpdate(resourceGroup.name(), eventGridTopic.name(),
+                    EVENT_SUBSCRIPTION_NAME, new EventSubscriptionInner()
+                        .withDestination(new EventHubEventSubscriptionDestination()
+                            .withResourceId(eventHub.id()))
+                        .withFilter(new EventSubscriptionFilter()
+                            .withIsSubjectCaseSensitive(false)
+                            .withSubjectBeginsWith("")
+                            .withSubjectEndsWith("")));
```

Comment:
Please use `eventGridManager.eventSubscriptions().define`:
```java

EventSubscription eventSubscription = eventGridManager.eventSubscriptions()
  .define(EVENT_SUBSCRIPTION_NAME)
  .withExistingScope(eventGridTopic.id())
  .withDestination(new EventHubEventSubscriptionDestination()
      .withResourceId(eventHub.id()))
  .withFilter(new EventSubscriptionFilter()
      .withIsSubjectCaseSensitive(false)
      .withSubjectBeginsWith("")
      .withSubjectEndsWith(""))
  .create();
```

Tested locally, it works.

Reply:
fixed by [7e4cbed](https://github.com/Azure/azure-sdk-for-java/pull/44748/commits/7e4cbed815f3d2199aab9db2104c273671e5ccfe)